core/src/main/java/io/grpc/internal/TransportSet2.java

```diff
@@ -182,9 +182,8 @@ final ClientTransport obtainActiveTransport() {
     Runnable runnable = null;
     synchronized (lock) {
       // Check again, since it could have changed before acquiring the lock
-      savedTransport = activeTransport;
-      if (savedTransport != null) {
-        return savedTransport;
+      if (activeTransport != null) {
```

Comment:
Still use a local variable, to avoid double-reading the volatile. Not a big deal, but good to do.